controllers/profiles/common/mutate_visitors.go

```diff
@@ -96,25 +100,47 @@ func EnsureDeployment(original *appsv1.Deployment, object *appsv1.Deployment) er
 	object.Spec.Replicas = original.Spec.Replicas
 	object.Spec.Selector = original.Spec.Selector
 	object.Labels = original.GetLabels()
+	object.Finalizers = original.Finalizers
 
 	// Clean up the volumes, they are inherited from original, additional are added by other visitors
-	object.Spec.Template.Spec.Volumes = nil
+	// However, the knative mount path must be preserved
+	var kneVol *corev1.Volume = nil
```

Comment:
Please keep this in a different mutator.

Reply:
I am not sure how this can be done in another mutator, because the knative volume is injected by Knative Eventing, not the SonataFlow operator.